fishtank.py

```diff
@@ -109,6 +116,33 @@ def move(self, walls, goblins):
             elif self.change_y < 0 and self.rect.top != goblin.rect.top:
                 self.rect.top = goblin.rect.bottom
 
+    def pick_target(self, possible_targets):
+        target_object = None
+
+        def look_within_cutoff(cutoff):
+            for target in possible_targets:
+                if abs(target.rect.x - self.rect.x) < cutoff and \
+                   abs(target.rect.y - self.rect.y) < cutoff:
+                    dist = distance(target.rect.x, target.rect.y, self.rect.x, self.rect.y)
+                    yield (dist, target)
+
+        for cutoff in (8, 128):
+            distances = look_within_cutoff(cutoff)
+            if distances:  # not empty
+                distances = sorted(distances)
+                try:
+                    target_object = distances[0][1]  # 0th (shortest dist), then the 1th element (object itself)
+                    break
+                except IndexError:
+                    continue
+
+        # too far away, just pick one at random
+        if target_object is None:
+            target_object = random.choice(list(possible_targets))
+
+        assert target_object is not None
+        return target_object
+
```

Comment:
Moved this to Organism. In theory Ogre can use it to find Goblins, but it doesn't do that yet.